### ISSUE TITLE
Using deprecated '-debug' fallback for parameter name resolution

### ISSUE THREAD

**Issue by Artur-** [vaadin/flow#15415]
### Description of the bug

Starting a Vaadin 24 app logs
```
2022-12-08T19:42:36.504+02:00  WARN 87982 --- [  restartedMain] ocalVariableTableParameterNameDiscoverer : Using deprecated '-debug' fallback for parameter name resolution. Compile the affected code with '-parameters' instead or avoid its introspection: com.vaadin.flow.spring.VaadinServletConfiguration
```

### Expected behavior

There should be no warning

### Minimal reproducible example

```
npx @vaadin/cli init --next hello
cd hello
mvn
```

### Versions

Vaadin: 24.0.0.alpha5
Flow: 24.0-SNAPSHOT
Java: Homebrew 19
OS: aarch64 Mac OS X 12.6.1
Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36
Live reload: Java active (Spring Boot Devtools): Front end active

**Comment by mcollovati** [comment]
May have been fixed in Spring 6.0.3 https://github.com/spring-projects/spring-framework/issues/29612

**Comment by Artur-** [comment]
This no longer reproduces
